src/analysis/souffle/dataflow_graph.dl

```diff
@@ -21,18 +21,49 @@
 // A data flow edge.
 .decl edge(src: AccessPath, tgt: AccessPath)
 
+// Use the previous relations to map an edge to a generated access path
+// representing its midpoint.
+.decl edgeToMidpointAccessPath(src: AccessPath, tgt: AccessPath, midpoint: AccessPath)
+
+// An "innerEdge" is an internal concept. When the user requests an edge with an
+// edge fact, we will expand that into innerEdges in one of two ways:
+//
+// 1. The edge is not the subject of a claimNotEdge, in which case an
+// innerEdge directly corresponding to the edge is created.
+//
+// 2. The edge is the subject of a claimNotEdge, in which case three facts
+// are created: a generated midway point AccessPath, an innerEdge going from the
+// src of the original edge to the midway point, and the midway point AccessPath
+// to the tgt. That allows us to effectively eliminate edges that are the
+// subject of a claimNotEdge from the dataflow graph by eliminating all tags
+// from the midway point. Paths are defined in terms of innerEdges.
+.decl innerEdge(src: AccessPath, tgt: AccessPath)
```

Comment:
`edgeInternal` (or `internalEdge`) might be better name?

Or, even something like `resolvedEdge` might work.

Reply:
Sure, I think those are all better. I was bothered by the potential confusion between "an edge inner to a particle" and "an edge internal to the Datalog script's reasoning about the world" after I committed this too. I like `resolvedEdge`, it clearly indicates that this is an edge that we have settled on after performing some kind of reasoning about `edge`s.

Reply:
Done.